include/NGenConfig.h.in

```diff
@@ -1,8 +1,57 @@
 #ifndef NGEN_NGENCONFIG_H
 #define NGEN_NGENCONFIG_H
 
+#define NGEN_STRINGIFY(x) #x
+#define NGEN_STRING(x) NGEN_STRINGIFY(x)
+
 #define ngen_VERSION_MAJOR @ngen_VERSION_MAJOR@
 #define ngen_VERSION_MINOR @ngen_VERSION_MINOR@
 #define ngen_VERSION_PATCH @ngen_VERSION_PATCH@
+#define ngen_VERSION NGEN_STRING(ngen_VERSION_MAJOR.ngen_VERSION_MINOR.ngen_VERSION_PATCH)
+
+#cmakedefine01 NGEN_WITH_MPI
+#cmakedefine01 NGEN_WITH_NETCDF
+#cmakedefine01 NGEN_WITH_SQLITE
+#cmakedefine01 NGEN_WITH_UDUNITS
+#cmakedefine01 NGEN_WITH_BMI_FORTRAN
+#cmakedefine01 NGEN_WITH_BMI_C
+#cmakedefine01 NGEN_WITH_PYTHON
+#cmakedefine01 NGEN_WITH_ROUTING
+#cmakedefine01 NGEN_WITH_TESTS
+#cmakedefine01 NGEN_QUIET
+
+#include <string>
+
+namespace ngen {
+
+struct exec_info
```

Comment:
Probably should add a delete constructor here just to make sure it's never (accidentally) initialized (which really shouldn't happen anyways, since there aren't any instance members)

Reply:
Just make it a `namespace`, then.

Reply:
If converted to a namespace, should the ns members be non-static? Since the variables are `constexpr` (and implicitly inline), and the functions are defined in `NGen.cpp` -- to avoid multiple copies of them in the binary?

Or is that mostly negligible?

Reply:
The variables would be fine. The functions should maybe be made non-`static`, so that attempts at calling them elsewhere than where they're defined in `NGen.cpp` won't lead to weird compilation errors about missing definitions.

Reply:
Added in [6734ba2](https://github.com/NOAA-OWP/ngen/pull/679/commits/6734ba213913c4da078fe63141253cc1b88eebc5).